apps/src/studio/studio.js

```diff
@@ -2643,12 +2702,23 @@ Studio.allGoalsVisited = function() {
     var goal = Studio.spriteGoals_[i];
     if (!goal.finished) {
       if (protagonistSprite) {
+        var wasGoalFinished = goal.finished;
+
         goal.finished = spriteAtGoal(protagonistSprite, goal);
+
+        // If goal was just finished, then call the "when actor touches anything handler"
+        if (! wasGoalFinished && goal.finished) {
+          var allowQueueExtension = false;
+          var prefix = 'whenSpriteCollided-' + Studio.protagonistSpriteIndex + '-';
+          callHandler(prefix + 'anything', allowQueueExtension);
+        }
+
```

Comment:
The previous level was like that, and the design is that in this level we
ask the user to score a point each time they touch the goal as well.


On Thu, Jan 22, 2015 at 6:16 PM, Bjvanminnen <notifications@github.com>
wrote:

> In apps/src/studio/studio.js
> <https://github.com/code-dot-org/code-dot-org/pull/1074#discussion_r23358635>
> :
>
> >          goal.finished = spriteAtGoal(protagonistSprite, goal);
> > +
> > +        // If goal was just finished, then call the "when actor touches anything handler"
> > +        if (! wasGoalFinished && goal.finished) {
> > +          var allowQueueExtension = false;
> > +          var prefix = 'whenSpriteCollided-' + Studio.protagonistSpriteIndex + '-';
> > +          callHandler(prefix + 'anything', allowQueueExtension);
> > +        }
> > +
>
> Nit: No space between ! and wasGoalFinished.
> It is most likely safe, but I'm not 100% sure. We could get into trouble
> if there are existing levels that have both goals and collide with anything
> events.
> Do we need to have score involved? It sounds like you're doing something
> like this level maybe? http://studio.code.org/s/playlab/stage/1/puzzle/6
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/1074/files#r23358635>.
>